tests/libgit/test_libgit_spec.lua

```diff
@@ -1,5 +1,25 @@
+-- Tests For luagit2's Libgit Module Functions.
+
 describe("Libgit2 core methods Test", function()
-  local lib , MWINDOW_SIZE , MWINDOW_MAPPED_LIMIT
+
+  local lib
+  local MWINDOW_SIZE = 1234
+  local MWINDOW_MAPPED_LIMIT = 1234
+  local Object_Cache_Limit = 1234
+  local Cache_Max_Size = 2048
+  local Allowed_Cache_memory , Current_Cache_Memory
+  local Template_path = "./Example/Template/path/"
+  local Windows_Share_Mode = 99999
+  local Enable = 1
+  local Disable = 0
+  local User_Agent = "Example_User_Agent"
+  local SSL_Cipher = "Example_SSL_Cipher"
+  local Search_Paths = {}
```

Comment:
in idiomatic Lua this would normally be written like:

```
local search_paths = {
        system_lvl = "./System_level/Search_Path/",
        global_lvl = "./Global_level/Search_Path/",
        xdg_lvl = "./XDG_level/Search_Path/",
        program_data_lvl = "./Program_data_level/Search_Path/",
}
```
with the table fields inside the table declaration. 
Concerning the capitalisation, it depends a lot. We normally all caps for consts and snake_case (all lower case with underscores) for the rest. There are situations we use PascalCase, but mostly only when dealing with prototypes. Exceptions are when libraries we use require a certain capitalisation. Does libgit2 require "Global_lvl" to be exactly like this?

Reply:
No, not at all, I just did this for readability . I'll do so if required.

Reply:
ok, then let's try to make it the more "Lua way" 
:)

Not everything in this list is mandatory, but here is a very good guide on Lua style: https://github.com/luarocks/lua-style-guide

Reply:
Thankyou . I will follow them :smile: